### PR TITLE
Reader: Add an ABTest for Reader Full Post, new flags

### PR DESCRIPTION
**Pull request by blowery**
This PR adds the ability to hide the new full post view behind an ABTest, in addition to a feature flag. If you enable the `reader/refresh/full-post-ab-test` feature flag, the full post abtest will become active. In addition, there's a feature flag, `reader/refresh/force-full-post` to force the new full post to be active, without turning on an abtest. 

It's split up this way to allow us to force the full post view on in certain environments and use the abtest in others. The new config included in this PR uses the abtest in dev and staging, but the force flag on wpcalypso and horizon. This way, we can test mobile, where it's hard to get into an abtest forcibly, on wpcalypso and/or horizon. And we can test the abtest on local dev boxes, or staging.

This PR also removes all of the old `reader-refresh` section work we were using to scope css and moves to the somewhat badly named `is-reader-page`, which is hooked up by the `ReaderMain` component. At the moment, only the new full post view uses `ReaderMain`, so it's an effective feature gate.

### CODE REVIEW

File: config/development.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
 	"logout_url": "https://wordpress.com/wp-login.php?action=logout&redirect_to=https%3A%2F%2F|subdomain|wordpress.com",
 	"facebook_api_key": "249643311490",
 	"discover_logged_out_redirect_url": "https://discover.wordpress.com",
+	"reader-full-post-abtest-entry-rate": 100,
```
Comment (bluefuton) [inline config/development.json]: Does this get used anywhere, or is it leftover?

**Comment by bluefuton** [comment]
All looks good, but I have to admit that `is-reader-page` confused me a bit. I imagine we don't want it to be something that refers to the refresh? Or 'new', because soon it won't be?

File: config/development.json (hunk 1 above)
Comment (blowery) [inline config/development.json]: leftover, good catch

**Comment by blowery** [comment]
> All looks good, but I have to admit that is-reader-page confused me a bit. I imagine we don't want it to be something that refers to the refresh? Or 'new', because soon it won't be?

Yeah, it's a bit odd. It's primary use is to make the background white and support the layout changes we need in Reader. Once the refresh is complete, we'll still need it and it'll be a "Reader" thing, hence the name. 

Maybe `is-editorial-page` since we're calling Reader-style stuff "editorial" design? Or is that even more obtuse?

**Comment by bluefuton** [comment]
I think we could end up with `is-editorial-page` being confusing one day - feels like it could be related only to Discover.

Names are hard. 

I think `is-reader-page` is workable and can't think of anything clearer - let's make sure its purpose is documented somewhere :) (p2?)

**Comment by blowery** [comment]
@bluefuton added a comment to ReaderMain to explain what it's all about

**Comment by bluefuton** [comment]
Works a treat on development :shipit:

**Pull request merged**